arch/xtensa/include/xtensa_asm2_s.h

```diff
@@ -604,6 +604,11 @@ _Level\LVL\()Vector:
 	s32i a2, a1, ___xtensa_irq_bsa_t_a2_OFFSET
 	s32i a3, a1, ___xtensa_irq_bsa_t_a3_OFFSET
 
+#ifdef CONFIG_ADSP_IDLE_CLOCK_GATING
+	/* Needed when waking from low-power waiti state */
+	isync
+#endif
```

Comment:
Is this the right place?  At this stage the handler has already significant code, including stores.

Normally you use an isync to prevent any in-flight state changes that might affect instruction fetch (i.e. MMU/MPU changes) from affecting the next instruction in the stream.    Which instruction are we trying to protect, and from what?  Is this maybe something that only happens if the ISR crosses a page/cacheline/whatever boundary?  If so that seems like it could be fixed with alignment instead?

My gut says that if we do need this, you almost certainly want this as the very first instruction in the vector, no?  Is there an errata or something that explains what you're trying to fix?

Reply:
We do need this @andyross - pls reach out to @kv2019i for further details.

Reply:
But right here?  It's just a really weird spot.

Reply:
@andyross Following up now to move this to a more maintainable place https://github.com/thesofproject/sof/pull/9031